solver/utils.go

```diff
@@ -14,6 +19,45 @@ var (
 	differ = gojsondiff.New()
 )
 
+func getDocumentDiff(a, b *state.Document) (string, error) {
+	aCopy := a.ShallowCopy()
+	bCopy := a.ShallowCopy()
+	aContent := *a.Content
+	bContent := *b.Content
+	aCopy.Content = nil
+	bCopy.Content = nil
+	objDiff, err := getDiff(aCopy, bCopy)
+	if err != nil {
+		return "", err
+	}
+	var contentDiff string
+	if json.Valid([]byte(aContent)) {
+		var aObj, bObj interface{}
+		err = json.Unmarshal([]byte(aContent), &aObj)
+		if err != nil {
+			return "", err
+		}
+		err = json.Unmarshal([]byte(bContent), &bObj)
+		if err != nil {
+			return "", err
+		}
+		aBytes, err := json.MarshalIndent(aObj, "", "\t")
+		if err != nil {
+			return "", err
+		}
+		bBytes, err := json.MarshalIndent(bObj, "", "\t")
+		if err != nil {
+			return "", err
+		}
+		aContent = string(aBytes)
+		bContent = string(bBytes)
+	}
+	edits := myers.ComputeEdits(span.URIFromPath("old"), aContent, bContent)
+	contentDiff = fmt.Sprint(gotextdiff.ToUnified("old", "new", aContent, edits))
+
+	return objDiff + contentDiff, nil
```

Comment:
Can we unit test this function?

Reply:
At the minimum, please abstract out the json content diff part into a separate function and unit test that.
This is a blocker for this merge.

Reply:
In the strictest sense of test, yes. 5327795 is uh... not great because multiline text wants are no fun.